Hazel.UnitTests/Dtls/DtlsConnectionTests.cs

```diff
@@ -413,17 +492,18 @@ public void TestResentHandshakeConnects()
                     listenerToConnectionThrottle.Release(1);
 
                     // Trigger a resend of ServerHello, ServerCertificate
+                    capture.DiscardPacketForLocal();
                     Thread.Sleep(1000);
-                    listenerToConnectionThrottle.Release(1);
+                    listenerToConnectionThrottle.Release(capture.PacketsForLocalCount);
 
                     // ServerKeyExchange, ServerHelloDone
                     listenerToConnectionThrottle.Release(1);
 
                     // Trigger a resend of ServerKeyExchange, ServerHelloDone
                     Thread.Sleep(1000);
-                    listenerToConnectionThrottle.Release(1);
 
                     capture.SendToLocalSemaphore = null;
+                    listenerToConnectionThrottle.Release(1);
```

Comment:
Curious why it is 1 here vs PacketsForLocalCount and also, was the order change here significant?

Reply:
The order change is no longer significant, but I do like it better. The intent at the bottom is to let the capture go freely, but if we release and capture sends one packet and loops, waiting again for release, then in a previous iteration, the thread would never stop waiting. In this iteration, though the 100ms timeout for responsiveness kicks in and things work normally. This ordering is tolerant of either, so I like it better.

Reply:
Ahh, cool, after reading deeper I see it now - I think actually this order is technically correct as if you release first, the thread that might of been waiting technically might wait again before this thread sets the reference to null

So yeah, this is better 😁